synapse/lib/stormtypes.py

```diff
@@ -8382,16 +8362,28 @@ async def _methTriggerDel(self, prefix):
 
         return iden
 
-    async def _methTriggerMod(self, prefix, query):
-        useriden = self.runt.user.iden
-        query = await tostr(query)
+    async def _methTriggerMod(self, prefix, edits):
         trig = await self._matchIdens(prefix)
         iden = trig.iden
-        gatekeys = ((useriden, ('trigger', 'set'), iden),)
-        todo = s_common.todo('setTriggerInfo', iden, 'storm', query)
-        await self.dyncall(trig.view.iden, todo, gatekeys=gatekeys)
+        edits = await toprim(edits)
 
-        return iden
+        viewiden = trig.view.iden
+        for name in edits:
+            self.runt.confirm(('trigger', 'set', name), gateiden=viewiden)
```

Comment:
Updating `user` is a special case for perms as they need that perm on the cortex rather than the view gate to prevent perms bypass.